### PR TITLE
TEXT-101: Moved RandomStringUtils to Commons-text 

### PR DESCRIPTION
**Pull request by ameyjadiye**
Moved **RandomStringUtils** from lang to text along with changed core logic which is using **RandomStringGenerator** internally, Test cases are copied from commons-lang for making sure its working as it's working in commons-lang.

### CODE REVIEW

**Comment by ameyjadiye** [comment]
@chtompki , can you please take look at this PR ?

**Comment by ameyjadiye** [comment]
@chtompki , can you take a look if this looks good ?

**Comment by chtompki** [comment]
Sure I'll get to this next.

File: src/main/java/org/apache/commons/text/RandomStringGenerator.java [hunk 1]
```diff
@@ -324,6 +324,40 @@ public Builder withinRange(final int minimumCodePoint, final int maximumCodePoin
...
+         * </pre>
+         *
+         * @param pairs array of charachters array, expected is to pass min, max pairs through this arg.
+         * @return {@code this}, to allow method chaining.
+         */
+        public Builder withinRange(final char[] ... pairs) {
+            characterList = new ArrayList<Character>();
+            for (char[] pair :  pairs) {
+                final int minimumCodePoint = pair[0];
+                final int maximumCodePoint = pair[1];
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/RandomStringGenerator.java]: Do we want validation on the definition of a `pair` here? Do we want to throw an `InvalidParameterException` in this fashion:
```java
for(char[] pair: pairs) {
    if (pair.length != 2) {
          throw new InvalidParameterException("A char[] pair should look like {'<char minimumCodePoint>', '<char maximumCodePoint>'}");
    }
}
I'm not sure. I could go either way. Thoughts?
```

Comment (ameyjadiye) [reply src/main/java/org/apache/commons/text/RandomStringGenerator.java]: I think we should put that for better exception handling, also check of ```minimumCodePoint  < maximumCodePoint ```, thanks

**Comment by ameyjadiye** [comment]
Hi @chtompki , do you see more places to improvement or its good now ?

**Comment by chtompki** [comment]
Should be able to look again today.

**Comment by chtompki** [comment]
As I'm pulling this in this crosses my mind: 

Wait I'm curious here. Weren't folks leaning towards keeping `RandomStringUtils` in `[lang]`?

The thread in question is this one: http://markmail.org/message/5tblwhldrwsbfdym

It feels like @britter, @PascalSchumacher, and @garydgregory were all leaning towards the `[lang]` option. If that's the case we should probably go with that.

**Comment by ameyjadiye** [comment]
@chtompki , I think all were thinking to deprecate it from lang and so we did. and due to more requests for ```RandomStringUtils```  we came up with the plan to move it also I think since ```RandomStringUtils``` is texty thing so belongs to commons-text and not lang. In same  mail trail you will see _"Grey area" should favour small components._

you should check this old mail trail as well http://markmail.org/message/bwyhvppr5s4waedp

**Comment by chtompki** [comment]
@britter, @PascalSchumacher, @garydgregory - thoughts?

**Comment by ameyjadiye** [comment]
As per discussion on Dev ML we are undoing deprecation of RandomStringUtils in Lang and discarding this solution.

**Pull request closed without merging**